### DISCUSSION TITLE
Selecting textlayer in a smoother way

### DISCUSSION THREAD

**Discussion by lfoppiano** [mozilla/pdf.js#18629] [General]
I'm working on a Streamlit pdf viewer using the pdf.js library programmatically, I've managed to do everything I wanted, show a PDF, show annotations, enable text layers, etc... 

However, I have a small problem when one try to select text in the PDF. It's a general issue I'm having using the text layers, as soon as the cursor get out of the text layer, the selection goes south. 

See an example in this screencast: 

https://github.com/user-attachments/assets/830d54ab-16b9-470c-816f-f1c4e8a5d0c5

The implementation is [here](https://github.com/lfoppiano/streamlit-pdf-viewer/blob/0034b681ddec9da981f6823ab6a9a73099614eea/streamlit_pdf_viewer/frontend/src/PdfViewer.vue#L124).

You can see a demo [here](https://structure-vision.streamlit.app/) (you must switch "enable text layers" on top-left). 

If you have any feedback or suggestion, It would be great. 

Thanks in advance

**Comment by radiachkik** [comment]
I am experiencing similar issues and would really appreciate some help. Is there something we can do in terms of configuration?

**Reply by lfoppiano** [reply]
I partially solve the issue by giving the text layer a higher z-index than the annotations, so that the text can be higlighted. However, I did not manage to get a much better effect.
